### PR TITLE
Enable log dump for NVMe suites

### PR DESCRIPTION
**Pull request by tintumathew10**
# Description

Enabling log dump for NVMe suites.

https://issues.redhat.com/browse/RHCEPHQE-16391

### CODE REVIEW

**Comment by tintumathew10** [comment]
logs - https://ocs3xstage-jenkins-csb-rhgsocs3x.apps.ocp-c1.prod.psi.redhat.com/view/Executor/job/qe-squid-test-executor-openstack/86/console

http://magna002.ceph.redhat.com/cephci-jenkins/results/openstack/IBM/8.0/rhel-9.4/Test/19.2.0-53/86/tier-1_nvmeof_sanity/

**Review by deepssin — DISMISSED** [review]
LGTM.

Have couple of thoughts that might improve this further -

1. Adding -z flag to tar command could help compress the archive. This would reduce storage usage and save network bandwidth during transfers hence fast downloads.
2. Another point to consider is to cleanup the tar from nodes after successful download so that the tests don't leave any residue on nodes.

Thanks!

**Comment by tintumathew10** [comment]
> LGTM.
> 
> Have couple of thoughts that might improve this further -
> 
> 1. Adding -z flag to tar command could help compress the archive. This would reduce storage usage and save network bandwidth during transfers hence fast downloads.
> 2. Another point to consider is to cleanup the tar from nodes after successful download so that the tests don't leave any residue on nodes.
> 
> Thanks!

Thanks Deepshika for the response.

Point 2 is taken care already.

File: cephci/cluster_info.py [hunk 1]
```diff
@@ -169,38 +168,16 @@ def get_ceph_var_logs(cluster, log_dir):
...
-                    src=os.path.join(CEPH_VAR_LOG_DIR, attribute.filename),
-                    dst=os.path.join(download_dir, attribute.filename),
-                    sudo=True,
-                )
-                log.info(
-                    f"Downloading {attribute.filename} from {node.hostname} to {download_dir}"
-                )
+        tar_file = "cephlog.tar"
+        node.exec_command(
+            sudo=True, cmd="tar -cvf %s %s" % (tar_file, CEPH_VAR_LOG_DIR)
```
Comment (psathyan) [inline cephci/cluster_info.py:173]: Can we `gzip`? That is 
```
node.exec_command(cmd=f"tar -cvzf {tar_file}, {CEPH_VAR_LOG_DIR}", sudo=True)
```

**Review by psathyan — COMMENTED** [review]
Thanks @tintumathew10 ... this does improve a lot of things. A minor change

File: cephci/cluster_info.py (hunk 1 above)
Comment (tintumathew10) [reply cephci/cluster_info.py]: done

**Review by deepssin — DISMISSED** [review]

File: cephci/cluster_info.py [hunk 2]
```diff
@@ -169,38 +168,15 @@ def get_ceph_var_logs(cluster, log_dir):
     for node in cluster.get_nodes():
         download_dir = os.path.join(log_dir, "ceph_logs", node.hostname)
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply cephci/cluster_info.py:169]: done

File: cephci/cluster_info.py [hunk 3]
```diff
@@ -169,38 +168,15 @@ def get_ceph_var_logs(cluster, log_dir):
...
-            elif S_ISREG(attribute.st_mode):
-                node.download_file(
-                    src=os.path.join(CEPH_VAR_LOG_DIR, attribute.filename),
-                    dst=os.path.join(download_dir, attribute.filename),
-                    sudo=True,
-                )
-                log.info(
-                    f"Downloading {attribute.filename} from {node.hostname} to {download_dir}"
-                )
+        tar_file = f"{node.hostname}-cephlog.tar"
```
[the remark this answers is not in the document]
Comment (tintumathew10) [reply cephci/cluster_info.py:171]: done

File: cephci/cluster_info.py [hunk 4]
```diff
@@ -167,40 +166,17 @@ def get_ceph_var_logs(cluster, log_dir):
     ceph cluster var logs into log directory.
     """
     for node in cluster.get_nodes():
-        download_dir = os.path.join(log_dir, "ceph_logs", node.hostname)
+        download_dir = os.path.join(log_dir, "ceph_logs")
```
Comment (vamahaja) [inline cephci/cluster_info.py:169]: `os.path.join` and `os.makedirs` are same across the loop. Move these steps out of loop

**Review by vamahaja — COMMENTED** [review]
LGTM.

Please share execution logs.

**Comment by tintumathew10** [comment]
https://ocs3xstage-jenkins-csb-rhgsocs3x.apps.ocp-c1.prod.psi.redhat.com/view/Executor/job/qe-squid-test-executor-openstack/89/console

http://magna002.ceph.redhat.com/cephci-jenkins/results/openstack/IBM/8.0/rhel-9.4/Test/19.2.0-53/89/tier-1_nvmeof_sanity/ceph_logs/

**Comment by tintumathew10** [comment]
> LGTM.
> 
> Please share execution logs.

logs attached

**Review by vamahaja — APPROVED** [review]

**Pull request merged**